test/unit/storage_test.rb

```diff
@@ -113,6 +113,35 @@ def test_sentinels_simple_url
                                       sentinels: [{ host: '127.0.0.1', port: 26_379 }])
   end
 
+  def test_sentinels_array_hashes_default_port
+    default_sentinel_port = Storage::Helpers.singleton_class.const_get(:DEFAULT_SENTINEL_PORT)
+    config_obj = {
+      url: 'redis://127.0.0.1:6379/0',
+      sentinels: [{ host: '127.0.0.1' }, { host: '192.168.1.1'},
+                  { host: '192.168.1.2', port: nil }]
+    }
+    conn = Storage.send :orig_new, Storage::Helpers.config_with(config_obj)
+    assert_sentinel_connector(conn.client)
+    assert_client_config(conn.client, url: config_obj[:url],
+                                      sentinels: [{ host: '127.0.0.1', port: default_sentinel_port },
+                                                  { host: '192.168.1.1', port: default_sentinel_port },
+                                                  { host: '192.168.1.2', port: default_sentinel_port }])
+  end
+
+  def test_sentinels_array_strings_default_port
+    default_sentinel_port = Storage::Helpers.singleton_class.const_get(:DEFAULT_SENTINEL_PORT)
+    config_obj = {
+      url: 'redis://127.0.0.1:6379/0',
+      sentinels: ['127.0.0.2', 'redis://127.0.0.1', '192.168.1.1']
```

Comment:
also: remember that when we _do_ specify sentinels then the url should contain the master group name - I think we don't check this and let the client complain, can you check?

Reply:
The idea is to let the Redis client crash, because the sentinel master group names is specified in a Redis URI too.

However, I think it's a good idea to use a name more similar to a real use case in our tests. A IP was specified instead of a master group name because I copied it from another existing test that performed other sentinels tests. I have added another commit that changes this and for all the previously existing sentinel-related tests previous to this PR that have a 'url' field and a 'sentinels' field.